app/api/auth/auth-options.ts

```diff
@@ -18,20 +21,24 @@ export const authOptions: NextAuthOptions = {
   jwt: { secret: process.env.JWT_COOKIE_SECRET },
   session: {
     strategy: "jwt",
-    maxAge: 30 * 60,
   },
 
   callbacks: {
     async redirect({ baseUrl }) {
       return baseUrl;
     },
     async jwt({ token, account }) {
+      const customToken = token as CustomToken;
       if (account) {
-        token.id_token = account.id_token;
-        token.access_token = account.access_token;
+        customToken.access_token = account.access_token;
+        customToken.refresh_token = account.refresh_token;
+        customToken.expiry = Date.now() + 30 * 60 * 1000; // expires after 30 minutes
```

Comment:
Expiry should be part of the response when you get a new token. So you should be able to avoid this hardcoded value.